lib/_tls_wrap.js

```diff
@@ -886,6 +867,115 @@ exports.createServer = function createServer(options, listener) {
 };
 
 
+Server.prototype.setSecureContext = function(options) {
+  if (options === null || typeof options !== 'object')
+    throw new ERR_INVALID_ARG_TYPE('options', 'Object', options);
+
+  if (options.pfx)
+    this.pfx = options.pfx;
+  else
+    this.pfx = undefined;
```

Comment:
It could get a bit messy with so many options, but the following pattern may save a bit of boilerplate code in this method...

```js
Server.prototype.setSecureContext = function({
  pfx = undefined,
  key = undefined,
  passphrase = undefined,
  cert = undefined,
  clientCertEngine = undefined,
  ca = undefined,
  secureProtocol = undefined,
  crl = undefined,
  ciphers = undefined,
  ecdhCurve = undefined,
  dhparam = undefined,
  honorCipherOrder = false,
  secureOptions,
  sessionIdContext
} = {}) {
  // ...
}
```

Not sure it's actually that much better tho ;-)

Reply:
@jasnell I like that pattern, because I think it would be helpful for code coverage purposes. But I'm a little worried that it's just slightly different enough from the existing options handling (`setOptions()`) to cause problems. For example, right now falsy values get filtered out in most cases, but that would no longer be the case. The current structure also allows us to make the validation stricter, because right now it seems rather loose.

I already plan to follow this up with a deprecation PR for `setOptions()`. I'd like to also revisit the argument validation here, but that will be semver-major.